Makefile

```diff
@@ -0,0 +1,28 @@
+SOURCE_FILES?=$$(glide novendor)
+TEST_PATTERN?=.
+TEST_OPTIONS?=
+
+setup: ## Install all the build and lint dependencies
+	curl https://glide.sh/get | sh
+	go get -u github.com/kisielk/errcheck
+	go get -u github.com/golang/lint/golint
+	glide install
+
+test: ## Run all the tests
+	go test $(TEST_OPTIONS) -cover $(SOURCE_FILES) -run $(TEST_PATTERN) -timeout=30s
+
+lint: ## Run all the linters
```

Comment:
I have changed the lint task so that it uses gometalinter. Nice tool! I've run into it a few months ago but then forgot about it. I will add all the linters you added to the travis script now!